mapper/mapper.go

```diff
@@ -625,6 +620,38 @@ func CreateCensusDatasetLandingPage(ctx context.Context, req *http.Request, base
 	return p
 }
 
+func mapFilterOutputDims(filter filter.Model, queryStrValues []string, path string) []sharedModel.Dimension {
+	dimensions := []sharedModel.Dimension{}
+	for _, dim := range filter.Dimensions {
+		pDim := sharedModel.Dimension{}
+		pDim.Title = dim.Label
+		pDim.ID = dim.ID
+		pDim.IsAreaType = helpers.IsBoolPtr(dim.IsAreaType)
+		pDim.TotalItems = len(dim.Options)
+		midFloor, midCeiling := getTruncationMidRange(pDim.TotalItems)
+
+		for i, opt := range dim.Options {
+			if pDim.TotalItems > 9 && !helpers.HasStringInSlice(pDim.ID, queryStrValues) {
+				if isTruncationIndex(i, midFloor, midCeiling, len(dim.Options)) {
+					pDim.Values = append(pDim.Values, opt)
+					pDim.IsTruncated = true
+				}
```

Comment:
Nested guard without a nested else to complement line 635 means that `Values` and `isTruncated` can be in an unknown state. Suggest setting these to a reasonable default before line 634, and overriding as necessary.

Reply:
Agreed about the potential unknown state. Will have to have a little think about this one as I don't want to be doing something like 
```go
pDim[i].Values = ""
pDim[i].IsTruncated = false
```
to then reassign the values
```go 
pDim[i].Values = append(pDim.Values, opt)
pDim[i].IsTruncated = true
```
An else might be better or some sort of `break`

Reply:
If the defaults are Go's zero values then, if I've understood that mechanism correctly, the code as it is would be fine but if the default for Values was some other string (say), then setting it first and then overwriting it in a special case is a perfectly reasonable option. It's not like overwriting the values incurs some massive penalty in storage space or computation time, so it's better to write the code to be easily followed by another developer than make the compiler's job a bit easier.